File: documentation/user/en/query/requirements/examples/facet/reference-summary-histograms.cs
EvitaResponse<ISealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
		Query(
			Collection("Product"),
			FilterBy(
				HierarchyWithin(
					"categories",
					AttributeEquals("code", "e-readers")
				),
				EntityLocaleEquals("en")
			),
			Require(
				Page(1, 5),
				EntityFetch(AttributeContent("code")),
				ReferenceSummaryOfReferenceWithHistograms(
					"parameterValues",
					FacetStatisticsDepth.Counts,
					EntityFetch(AttributeContent("code")),
					EntityGroupFetch(AttributeContent("code")),
					HistogramStatistics(20, "intervalParameterValues")
				)
			)
		)
	)
);

File: documentation/user/en/query/requirements/examples/facet/reference-summary-histograms.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					hierarchyWithin(
						"categories",
						attributeEquals("code", "e-readers")
					),
					entityLocaleEquals(Locale.forLanguageTag("en"))
				),
				require(
					page(1, 5),
					entityFetch(attributeContent("code")),
					referenceSummaryOfReferenceWithHistograms(
						"parameterValues",
						FacetStatisticsDepth.COUNTS,
						entityFetch(attributeContent("code")),
						entityGroupFetch(attributeContent("code")),
						histogramStatistics(20, "intervalParameterValues")
					)
				)
			)
		);
	}
);
